WcaOnRails/app/models/user.rb

```diff
@@ -731,6 +731,12 @@ def can_edit_registration?(registration)
     can_manage_competition?(registration.competition) || (registration.user_id == self.id && editable_by_user)
   end
 
+  def can_delete_registration?(registration)
+    # A registration can only be deleted by a user after it has been accepted if the organizers allow
+    can_delete_by_user = (!registration.accepted? || registration.competition.registration_delete_after_acceptance_allowed?)
+    registration.user_id == self.id && can_delete_by_user
```

Comment:
⚠️ It is imperative that we always allow organisers/admins to delete any registration, regardless of how the competition is configured!

```suggestion
    can_manage_competition?(registration.competition) || (registration.user_id == self.id && can_delete_by_user)
```

Reply:
Your outline helped a lot with the speed!

For this though, admins are still able to delete registration from the admin registration list, and the new variable does not interfere with that. It only interferes with users deleting registration from the register form.

Reply:
Okay, makes sense in terms of functionality. I'd still argue for having a consistent codebase where the concrete implementation of methods does not change depending on which frontend views these methods are accessed from.

Unifying it would also mean you could (probably?) redirect the admin check to this new deletion function as well.

Reply:
I see, that makes sense.